client/me/purchases/manage-purchase/index.tsx

```diff
@@ -639,6 +642,76 @@ class ManagePurchase extends Component<
 		return null;
 	}
 
+	renderRefundText() {
+		const { purchase, translate } = this.props;
+
+		// Hide if refund window has lapsed.
+		if ( ! purchase?.isRefundable || ! purchase?.refundText || ! purchase?.mostRecentRenewDate ) {
+			return;
+		}
+
+		const ONE_DAY_IN_MILLISECONDS = 24 * 60 * 60 * 1000;
+		const lastRenewalDate = new Date( purchase?.mostRecentRenewDate );
+		const refundPeriodEndDate = new Date(
+			lastRenewalDate.getTime() + purchase?.refundPeriodInDays * ONE_DAY_IN_MILLISECONDS
+		);
+
+		const refundText = (() => {
+			// User would receive another free domain credit upon cancelling.
+			if ( isDomainRegistration( purchase ) && ! hasAmountAvailableToRefund( purchase ) ) {
+			  return translate(
+				"If you're unsure about this domain, cancel it before {{refundDate/}} to choose another free domain instead.",
+				{
+					components: {
+						refundDate: <FormattedDate date={ refundPeriodEndDate } format="LL" />,
+					},
+				}
+			);
+			}
+		  
+			// User would receive a financial refund upon cancelling.
+			return translate(
+				'Eligible for a refund of %(refundAmount)s until {{refundDate/}}.',
+				{
+					components: {
+						refundDate: <FormattedDate date={ refundPeriodEndDate } format="LL" />,
+					},
+					args: {
+						refundAmount: purchase?.refundText,
```

Comment:
Same here. We don't want to allow displaying `Elligible for a refund of undefined`.

```suggestion
						refundAmount: purchase.refundText,
```